lib/pbench/agent/results.py

```diff
@@ -149,23 +149,25 @@ class CopyResultTb:
     """CopyResultTb - Use the server's HTTP PUT method to upload a tarball
     """
 
-    chunk_size = 4096
+    CHUNK_SIZE = 65536
 
     def __init__(
         self, controller: str, tarball: str, config: PbenchAgentConfig, logger: Logger
     ):
+        self.controller = secure_filename(controller)
```

Comment:
You know, I've always kinda taken this `secure_filename` thing for granted, but seeing it on what's really "just a string" made me think about it. What we really want here is an HTML-sanitized string. While `secure_filename`'s "pure ascii" signature is similar, I wonder if this is really exactly what we want? Not a big deal here, I think, in any case ...

Reply:
["The function might return an empty filename."](https://tedboy.github.io/flask/generated/werkzeug.secure_filename.html)

I'm sure it only happens for edge cases, but...we don't check for it....

Reply:
What we probably should be doing is wrapping `secure_filename()` in a utility function:  this would allow us to do things like check it before we return it, and even completely replace the underlying mechanism, without affecting the calling code.

Reply:
We really don't need to use `secure_filename` on the agent side.  We only need that on the server side.  We have a secure file name on the agent side because we derive it from a file system.

We do need to `urllib.parse.quote()` it.